src/components/menu/menu.tsx

```diff
@@ -72,6 +96,15 @@ export class Menu {
     @Prop({ reflect: true })
     public gridLayout = false;
 
+    @Prop()
+    public loadSubItems: SubItemsLoader;
+
+    @Prop()
+    public lazyLoadItems: boolean;
+
+    @Prop()
+    public loading: boolean;
```

Comment:
There's quite lot of new functionality being introduced all at once here. Are these all necessary for the sub-menu functionality, or are they separate features that just happen to become want-to-have's when we add the sub-menu feature? Could we split this PR down a bit?

I noticed that on the `Files` tab, the diff for `menu.tsx` doesn't even render unless you click it, because it's so big. That's a little frightening… 😅

Reply:
Should this be a public prop? :thinking: I'm not quite sure how it should be used since there are no docs for it, but maybe it should be a private state of the component instead? :thinking:

Reply:
We want the end user to be able to show a loader in the menu surface if they for example loads the root items dynamically.
Our hierarchy picker (why we did this PR in the beginning) for example loads the root items from limeobjects that is async, and instead of just an empty menu we would like to support the loader in the menu.

Does it make sense?

Reply:
It would be possible to remove the loadingSubItems, but then the consumer have to control when the component is in a loading state form outside.